src/physics/base/PhysicsParams.cc

```diff
@@ -0,0 +1,62 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file PhysicsParams.cc
+//---------------------------------------------------------------------------//
+#include "PhysicsParams.hh"
+
+#include <algorithm>
+#include <iterator>
+#include "base/Range.hh"
+#include "ParticleParams.hh"
+#include "physics/material/MaterialParams.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with processes and helper classes.
+ */
+PhysicsParams::PhysicsParams(Input inp) : processes_(std::move(inp.processes))
+{
+    CELER_EXPECT(!processes_.empty());
+    CELER_EXPECT(std::all_of(processes_.begin(),
+                             processes_.end(),
+                             [](const SPConstProcess& p) { return bool(p); }));
+    CELER_EXPECT(inp.particles);
+    CELER_EXPECT(inp.materials);
+
+    // Resize: processes for each particle
+    processes_by_particle_.resize(inp.particles->size());
+
+    // Construct models, assigning each model ID
+    ModelIdGenerator next_model_id;
+    for (auto process_idx : range<ProcessId::value_type>(processes_.size()))
```

Comment:
Why not?:
```
for (auto& process : processes_.size())
```

Reply:
This part is all incomplete code sos it does nothing regardless -- but each process needs a corresponding `process_id` so looping over the `processes` directly vs indexing them is six of one, half a dozen of the other. Once it's implemented (see the todos on #125 ) I'll make sure it's more sensible :)